File: 04_counting_elements/perm_check.py
def solution(A):
  n = len(A)
  lst = [0] * (n + 1)
  for i in A:
    ind = min(i, n + 1)
    lst[ind-1] += 1
  for i in lst[:-1]:
    if i == 0:
      return 0
  return 1

File: 04_counting_elements/perm_check.cpp
#include <vector>

using namespace std;

int solution(vector<int> &a) {
  int n = a.size();
  vector<int> check(n+1, 0);
  int ret = 1;
  for (int i = 0; i < n; i++) {
    if (a[i] < n+1)
      check[a[i]] = 1;
  }
  for (int i = 1; i < n+1; i++) {
    ret *= check[i];
  }
  return ret;
}
